charts/gardener-extension-provider-aws/templates/deployment.yaml

```diff
@@ -9,6 +9,7 @@ metadata:
 {{- end }}
   labels:
 {{ include "labels" . | indent 4 }}
+    high-availability-config.resources.gardener.cloud/type: controller
```

Comment:
Short question :)
This component is both a controller and a server.
I'm wondering, how to deal with such cases. Is the plan to always use `controller` in such cases?

Reply:
We should rather use `server` for such cases (even though, right now it doesn't make a difference technically, but in the future `server`s might have more replicas than controllers for better network traffic control).

Reply:
Thanks for pointing this out. Makes sense 👍

Reply:
FYI: I filed https://github.com/gardener/gardener/pull/7060 to include this lessons learned.